FreeRTOS/Lab-Project-FreeRTOS-FAT
# Feature/fixing clang gnu compiler warnings

Fixing Clang and GNU compiler warnings along with some clang-tidy issues.

Description
-----------
* Upkeep/maintenance - fixed a bunch of the compiler warnings that were previously ignored. See lists in CMakeLists.txt for ones fixed.

Test Steps
-----------
* None - re-run test suite.
* 
Checklist:
----------
- [x ] I have tested my changes. No regression in existing tests.
- [x ] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Related Issue
-----------
None.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
